modules/swagger-codegen/src/main/resources/lua/api.mustache

```diff
@@ -0,0 +1,113 @@
+{{>partial_header}}
+//package {{packageName}}
+
+{{#operations}}
+local http_request = require "http.request"
+local http_util = require "http.util"
+local dkjson = require "dkjson"
+
+// model import
+local {{{packagName}}}_{{classname}} = require "{{{pacakgeName}}}.{{classname}}"
+
+local {{{packageName}}}= {}
+local {{{pacakgeName}}}_mt = {
+  __name = "{{{packageNameHyphen}}}";
+  __index = {{{packageName}}};
+}
+
+local function new_{{classname}}(host, basePath, schemes)
+  local schemes_map = {}
+  for _,v in ipairs(schemes) do
+    schemes_map[v] = v
+  end
+  local default_scheme = schemes_map.https or schemes_map.http
+  // TODO: default basePath to {{{basePath}}}
+  return setmetatable({
+    host = host;
+    basePath = basePath;
+    schemes = schemes_map;
+    default_scheme = default_scheme;
+  }, {{{pacakgeName}}}_mt)
+end
+
+{{#operation}}
+function {{classname}}:{{operationId}}({{#allParams}}{{#required}}{{paramName}}{{#hasMore}}, {{/hasMore}}{{/required}}{{/allParams}})
+    local req = http_request.new_from_uri({
+        scheme = self.default_scheme;
+        host = self.host;
+        path = string.format("%s/{{{path}}}", self.basePath, petId);
+    })
+
+    // set HTTP verb
+	req.headers:upsert(":method", "{{httpMethod}}")
+
+    // TODO: create a function to select proper accept
+    local var_accept = { {{#consumes}}"{{{mediaType}}}"{{#hasMore}}, {{/hasMore}}{{/consumes}} }
+    req.headers:upsert("accept", {{#consumes}}{{#-first}}"{{{mediaType}}}"{{/-first}}{{/consumes}})
+
+    // TODO: create a function to select proper content-type
```

Comment:
Here is an example for Java (jersey2):

https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/main/resources/Java/libraries/jersey2/ApiClient.mustache#L435-L491